decidim-proposals/lib/decidim/proposals/component.rb

```diff
@@ -265,7 +265,20 @@
         visibility: "all"
       ) do
         proposal = Decidim::Proposals::Proposal.new(params)
+
         proposal.add_coauthor(participatory_space.organization)
+
+        coauthor = case n
+                   when 0
+                     Decidim::User.where(decidim_organization_id: participatory_space.decidim_organization_id).order(Arel.sql("RANDOM()")).first
+                   when 1
+                     Decidim::UserGroup.where(decidim_organization_id: participatory_space.decidim_organization_id).order(Arel.sql("RANDOM()")).first
+                   when 2
+                     Decidim::Meetings::Meeting.where(component: component).order(Arel.sql("RANDOM()")).first
```

Comment:
I get a `nil` here and gives me an error while executing the seeds:
```ruby

bin/rails db:seed
Creating seeds for decidim-core...
xdg-open file:////home/apereira/Work/decidim/decidim/development_app/tmp/letter_opener/1643103311_6821141_5d3ab90/rich.html
Creating seeds for decidim-system...
Creating seeds for decidim-templates...
Creating seeds for the participatory_processes space...
-- Creating pages component seeds for the participatory space with ID: 1...
-- Creating meetings component seeds for the participatory space with ID: 1...
-- Creating proposals component seeds for the participatory space with ID: 1...
rails aborted!
NoMethodError: undefined method `id' for nil:NilClass
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/coauthorable.rb:143:in `add_coauthor'
/home/apereira/Work/decidim/decidim/decidim-proposals/lib/decidim/proposals/component.rb:281:in `block (4 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-core/app/services/decidim/traceability.rb:64:in `block (2 levels) in perform_action!'
/home/apereira/Work/decidim/decidim/decidim-core/app/services/decidim/traceability.rb:63:in `block in perform_action!'
/home/apereira/Work/decidim/decidim/decidim-core/app/services/decidim/traceability.rb:62:in `perform_action!'
/home/apereira/Work/decidim/decidim/decidim-proposals/lib/decidim/proposals/component.rb:261:in `block (3 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-proposals/lib/decidim/proposals/component.rb:235:in `times'
/home/apereira/Work/decidim/decidim/decidim-proposals/lib/decidim/proposals/component.rb:235:in `block (2 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/component_manifest.rb:137:in `seed!'
/home/apereira/Work/decidim/decidim/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb:248:in `block (4 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb:247:in `block (3 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb:107:in `times'
/home/apereira/Work/decidim/decidim/decidim-participatory_processes/lib/decidim/participatory_processes/participatory_space.rb:107:in `block (2 levels) in <main>'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/participatory_space_manifest.rb:94:in `seed!'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/core.rb:117:in `block in seed!'
/home/apereira/Work/decidim/decidim/decidim-core/lib/decidim/core.rb:116:in `seed!'
/home/apereira/Work/decidim/decidim/development_app/db/seeds.rb:9:in `<main>'
bin/rails:4:in `<main>'
Tasks: TOP => db:seed
(See full trace by running task with --trace)


```

It's because the `Decidim::Meetings::Meeting.where(component: component)`... Technically, we're talking about two different components in the same participatory space (one component for Proposals and another one for Meetings). I'd copy and adapt the code for finding the meetings from the proposals form (`app/forms/decidim/proposals/admin/proposal_base_form.rb`)

Tip: to debug it, I added a `debugger` invocation in the same line that has given the exception (281) and then I executed and checked the `coathor` value.

Reply:
Good point, that's my fault. Thanks for pointing it out

Reply:
No problem!